test/router_xor_filter_SUITE.erl

```diff
@@ -440,6 +442,104 @@ migrate_filter_test(Config) ->
     meck:unload(blockchain_worker),
     ok.
 
+move_to_front_test(Config) ->
+    %% When filters get large, they are expensive to add single devices to,
+    %% instead of manually creating migrations, we can say across what number of
+    %% filters we want to distribute the devices we know about.
+    %% -------
+    %% > router_xor_filter_worker:move_to_front(_NumGroups = 2, _Commit = true).
+    %% 0 - long term storage
+    %% 1 - long term storage
+    %% 2 - cheap adding devices
+    %% 3 - cheap adding devices
+    %% 4 - cheap adding devices
+    Chain = proplists:get_value(chain, Config),
+    OUI1 = proplists:get_value(oui, Config),
+    Tab = proplists:get_value(ets, Config),
+
+    %% Init worker
+    application:set_env(router, router_xor_filter_worker, false),
+    erlang:whereis(router_xor_filter_worker) ! post_init,
+
+    %% Wait until xor filter worker started properly
+    ok = test_utils:wait_until(fun() ->
+        State = sys:get_state(router_xor_filter_worker),
+        State#state.chain =/= undefined andalso
+            State#state.oui =/= undefined
+    end),
+
+    %% ------------------------------------------------------------
+    Round1Devices = n_rand_devices(10),
+    Round2Devices = Round1Devices ++ n_rand_devices(200),
+    Round3Devices = Round2Devices ++ n_rand_devices(30),
+    Round4Devices = Round3Devices ++ n_rand_devices(6),
+    Round5Devices = Round4Devices ++ n_rand_devices(50),
+
+    lists:foreach(
+        fun(#{devices := Devices, block := ExpectedBlock, filter_count := ExpectedFilterNum}) ->
+            true = ets:insert(Tab, {devices, Devices}),
+            ok = router_xor_filter_worker:check_filters(),
+
+            %% should have pushed a new filter to the chain
+            ok = expect_block(ExpectedBlock, Chain),
+
+            Filters1 = get_filters(Chain, OUI1),
+            ?assertEqual(ExpectedFilterNum, erlang:length(Filters1))
+        end,
+        [
+            #{devices => Round1Devices, block => 3, filter_count => 2},
+            %% largest filter
+            #{devices => Round2Devices, block => 4, filter_count => 3},
+            #{devices => Round3Devices, block => 5, filter_count => 4},
+            #{devices => Round4Devices, block => 6, filter_count => 5},
+            %% We should not craft filters above 5
```

Comment:
Would you annotate this comment to indicate why the limit should be 5, please?  (You explained it to me yesterday, but I've already become uncertain about the reason.)